Examples/Scripts/MaterialMapping/material_comparison.py

```diff
@@ -0,0 +1,239 @@
+    p.add_argument(
+        "-v",
+        "--variables",
+        type=str,
+        nargs="+",
+        default=["t_X0", "t_L0"],
+        help="The variables to be plotted",
+    )
+    p.add_argument(
+        "-m",
+        "--max",
+        type=float,
+        nargs="+",
+        default=[4.5, 1.5],
+        help="The variables to be plotted",
+    )
+    p.add_argument(
+        "-a",
+        "--axes",
+        type=str,
+        nargs="+",
+        default=["v_eta", "v_phi"],
+        help="The axes versus which the variables to be plotted",
+    )
+    p.add_argument(
+        "--eta",
+        type=float,
+        nargs=2,
+        default=[-3.0, 3.0],
+        help="Eta range for the plotting",
+    )
+    p.add_argument("--eta-bins", type=int, default=60, help="Eta bins for the plotting")
+    p.add_argument(
+        "--phi",
+        type=float,
+        nargs=2,
+        default=[-math.pi, math.pi],
+        help="Phi range for the plotting",
+    )
+    p.add_argument("--phi-bins", type=int, default=72, help="Phi bins for the plotting")
+    p.add_argument(
+        "-d",
+        "--detray-input",
+        type=str,
+        default="",
+        help="Optional detray input csv file",
+    )
+
+    args = p.parse_args()
+
+    ttrees = []
+
+    if len(args.input) != len(args.labels):
+        print("** ERROR ** The number of input files and labels must match")
+        exit(1)
+
+    # The histogram map
+    h_dict = {}
+    tfiles = []
+
+    ofile = ROOT.TFile.Open(args.output, "RECREATE") if args.output else None
+
+    input_files = args.input
+
+    # Detray csv file
+    if args.detray_input:
+        input_files.append(args.detray_input)
+
+    # Loop over the files and create the comparison histograms
+    for fi, ifile in enumerate(args.input):
+        # Special treament for detray input
+        if ifile == args.detray_input:
+            print("Reading detray input from file: ", args.detray_input)
+            ttree = TTree("t", "material read from detray")
+            ttree.ReadFile(args.detray_input, "v_eta/D:v_phi:t_X0:t_L0:t_X0:t_L0")
+            lbl = "detray"
+        else:
+            # Get the file
+            tfile = ROOT.TFile.Open(ifile)
+            tfiles.append(tfile)
+            ttree = tfile.Get("material-tracks")
+            ttrees.append(ttree)
+            # The label
+            lbl = args.labels[fi]
+
+        # Loop over the variables and axes
+        for iv, var in enumerate(args.variables):
+            for ax in args.axes:
+                if ax == "v_eta":
+                    bins = args.eta_bins
+                    low = args.eta[0]
+                    high = args.eta[1]
+                    cut = f"v_phi > {args.phi[0]} && v_phi < {args.phi[1]}"
+                elif ax == "v_phi":
+                    bins = args.phi_bins
+                    low = args.phi[0]
+                    high = args.phi[1]
+                    cut = f"v_eta > {args.eta[0]} && v_eta < {args.eta[1]}"
```

Comment:
I think you are inverting eta and phi in the cuts ? Or maybe I am missing something

Reply:
It's correct as is, if you look at the phi projection, you want to restrict eta to its range, such that you don't look at the phi projection outside your eta restriction.

Reply:
Ah yes you are right !